src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private int size;
+    private T[] elementsData;
+
+    public ArrayList() {
+        elementsData = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementsData.length) {
+            resize();
+        }
+        elementsData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            trowException();
+        }
+        if (size == elementsData.length - 1) {
```

Comment:
why in line 17  we check ` if (size == elementsData.length) `, but here is ` if (size == elementsData.length - 1)`?
also for the case when  `(index == size) ` you can simple call existing method `void add(T value)`